HPC/HPC.md

```diff
@@ -0,0 +1,24 @@
+# Getting started with HPC workloads on parallel clusters using Graviton CPUs
+
+**Introduction**
+
+HPC is a field where either high throughput and low latency is desired. Arm processor has its unique advantage in power efficiency and can find wide applications in HPC filed. Here is the instruction on how to get started with HPC on parallel clusters using Graviton CPUs.
+
+**How to get started running HPC workload on Graviton CPUs**
+First, follow the instructions here to setup a parallel cluster - https://docs.aws.amazon.com/parallelcluster/latest/ug/install-v3-configuring.html. OpenMPI is the default MPI and already installed on AWS parallel cluster. For compilers, arm compiler for linux and gcc 10.2 or later are recommended for better auto-vectorization. Arm performance library includes a dense and sparse linear algebra library, fft library and optimized math library that can also improve the throughput of HPC applications.  The arm compiler and library can be downloaded at https://developer.arm.com/downloads/-/arm-compiler-for-linux. 
+
+**One simple example - HPCCG** 
+- fetch the HPCCG code with arm64 support (from fork until PR is accepted): https://github.com/juntangc/HPCCG.
+
+- The recommended flags for compilation is "-Ofast -mcpu=native". 
```

Comment:
can we really say Ofast is ever recommended. I don't see how we can.

Reply:
I changed the recommendation to O3.

I would say the recommended flags (-Ofast -mcpu=native -funroll-loops -fno-expensive-optimizations -fno-reciprocal-math -fsigned-zeros -fno-unsafe-math-optimizations) used for WRF simulation is very safe because we managed to achieve bit-by-bit matching results on another architecture (x86_64 -O2).  

For HPCCG, it is computing conjugate gradient in double precision so Ofast won't be an issue.